src/lqcontrol.jl

```diff
@@ -16,90 +16,66 @@ http://quant-econ.net/lqcontrol.html
 =#
 
 type LQ
-    Q::Matrix
-    R::Matrix
-    A::Matrix
-    B::Matrix
-    C::Union(Nothing, Matrix)
+    Q::ScalarOrArray
+    R::ScalarOrArray
+    A::ScalarOrArray
+    B::ScalarOrArray
+    C::ScalarOrArray
+    N::ScalarOrArray
     bet::Real
-    T::Union(Int, Nothing)
-    Rf::Matrix
-    k::Int
-    n::Int
-    j::Int
-    P::Matrix
+    term::Union(Int, Nothing) # terminal period
+    rf::ScalarOrArray
```

Comment:
I see that we moved from `Rf` to `rf`. I know the convention is to have CamelCase for type names, and lower case for variable (field) names.

Sometimes, however, I think it is ok to match the notation of the model (like we do leaving `P` and `F` capitalized).

Any thoughts on this?